### PR TITLE
[#1789] First pass at adding preview text to trackevents

### PR DESCRIPTION
**Pull request by dseif**

### CODE REVIEW

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @jbuck received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/741f38e214d71d4/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @jbuck received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/688121719cf8fde/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/741f38e214d71d4/output.txt

Total script time: 0.90 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/688121719cf8fde/output.txt

Total script time: 1.00 mins

Server is running on http://pr709.butyr.org/templates/basic/

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @dseif received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/d6f9eb90ac2418a/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/d6f9eb90ac2418a/output.txt

Total script time: 0.68 mins

Server is running on http://pr709.butyr.org/templates/basic/

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @dseif received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/a7b48cd11f7bf98/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/a7b48cd11f7bf98/output.txt

Total script time: 1.23 mins

Server is running on http://pr709.butyr.org/templates/basic/

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @jbuck received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/a143eb62bb6bcf1/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/a143eb62bb6bcf1/output.txt

Total script time: 1.28 mins

File: test/trackevent/trackevent-preview-text.html [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+          butter.listen( "trackeventupdated", function( e ) {
+            equal( previewStrings.pop(), e.data.popcornOptions.text, "TrackEvent preview text is set correctly" );
+            if ( !previewStrings.length ) {
+              butter.unlisten( "trackeventupdated" );
+              start();
+            } else {
+              te.update({
+                  start: 5,
+                  end: 10,
+                  text: previewStrings[ 0 ]
```
Comment (mjschranz) [inline test/trackevent/trackevent-preview-text.html]: above 3 lines need to be over to the left by 2 spaces

**Comment by jbuck** [comment]
Rebased on master: https://github.com/mozilla/butter/commit/13f3f2c510e85a480de0ef442b5545b1674b046a

**Pull request closed without merging**